### PR TITLE
#3164: enable React strict mode in development mode

### PR DESCRIPTION
**Pull request by twschiller**
## What does this PR do?

- Some prep work for #3164
- Enables React strict mode in development mode for the lulz
- The goal is to prevent developers from introducing new `useEffect`/render bugs during development

## Discussion

- Guide recommends putting in on the render call vs. the root component: https://react.dev/reference/react/StrictMode#enabling-strict-mode-for-entire-app
- React recommends keeping double-analytics calls in development mode: https://react.dev/learn/synchronizing-with-effects#sending-analytics
- NOTE: our E2E CI tests do not run in development mode

## Remaining Work

- [x] Fix broken Jest tests
  - [x] React Select id sequence
  - [x] Fix render count assertions in tests
- [x] Smoke test local build
- [x] Double-check which runtime bugs found in https://github.com/pixiebrix/pixiebrix-extension/pull/9394 existing on this branch (e.g., workshop editor infinite render)
- [x] ~~Fix Page Editor horizontal width/scrollbar regression when the Chrome Side Panel is open (also see in https://github.com/pixiebrix/pixiebrix-extension/pull/9394)~~: the bug also exists on main

## Future Work

- Address StrictMode deprecation warning as needed. ⚠️ It's in react-bootstrap 4's `OverlayTrigger`: https://github.com/react-bootstrap/react-bootstrap/issues/5519. It's unclear if/when the feature will be retired:
> consoleObservable.js:33 Warning: findDOMNode is deprecated in StrictMode. findDOMNode was passed an instance of RefHolder which is inside StrictMode. Instead, add a ref directly to the element you want to reference. Learn more about using refs safely here: https://reactjs.org/link/strict-mode-find-node
- Consider making E2E CI tests run in development mode

For more information on our expectations for the PR process, see the
[code review principles doc](https://www.notion.so/pixiebrix/Code-Review-Principles-1ce7276b82a84d2a995d55ad85e1310d?pvs=4)

### CODE REVIEW

**Comment by grahamlangford** [comment]
> Enables React strict mode in development mode for the lulz

image

File: src/components/form/widgets/RemoteSelectWidget.test.tsx [hunk 1]
```diff
@@ -173,7 +173,10 @@ describe("RemoteSelectWidget", () => {
     await waitFor(() => {
       expect(setLocalErrorSpy).toHaveBeenCalledWith(null);
     });
-    expect(reportErrorSpy).toHaveBeenCalledOnce();
+
+    // React team recommends keeping duplicate analytics calls in development mode:
+    // https://react.dev/learn/synchronizing-with-effects#sending-analytics
```
Comment (twschiller) [inline src/components/form/widgets/RemoteSelectWidget.test.tsx:178]: See React guidance on analytics

File: src/telemetry/trace.ts [hunk 2]
```diff
@@ -341,12 +341,6 @@ export async function clearModComponentTraces(
         cnt++;
         await cursor.delete();
       }
-
-      console.debug(
```
Comment (twschiller) [inline src/telemetry/trace.ts:345]: Remove noisy debug code

File: src/pageEditor/tabs/editTab/dataPanel/DataTabJsonTree.test.tsx [hunk 3]
```diff
@@ -74,27 +74,49 @@ test("renders the DataTabJsonTree component", async () => {
...
-  // The expanded state in Redux has been updated, this triggers a re-render of the DataTabJsonTree and hence the JSONTree
-  expect((renderCount.current.JSONTree as RenderCountField).value).toBe(2);
+  // The expanded state in Redux has been updated, triggering a re-render of the DataTabJsonTree and hence the JSONTree
+  // Vendored react-performance-testing library not supported in React 18 (exact limitations unclear):
+  // https://github.com/keiya01/react-performance-testing/issues/27
+  expect(
+    renderCount.current.DataTabJsonTree as RenderCountField[],
+  ).toStrictEqual([
+    { value: 1 },
+    // XXX: why is this count 3x instead of 2x? Which StrictMode behavior causes the 2x when expanding the node?
```
Comment (twschiller) [inline src/pageEditor/tabs/editTab/dataPanel/DataTabJsonTree.test.tsx:112]: See interaction with react performance testing library

**Review by grahamlangford — APPROVED** [review]

**Pull request merged**